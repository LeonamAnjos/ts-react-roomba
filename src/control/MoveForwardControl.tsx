import React, { useContext } from "react";
import { RoombaContext } from "../contexts/RoombaContext";

const MoveForwardControl = () => {
  const { moveForward } = useContext(RoombaContext);

  return <button onClick={moveForward}>Move Forward</button>;
};

export default MoveForwardControl;
