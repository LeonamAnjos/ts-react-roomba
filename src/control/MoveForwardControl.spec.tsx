import { render, screen, fireEvent } from "@testing-library/react";
import { ReactNode, useContext } from "react";
import RoombaContextProvider, {
  RoombaContext,
} from "../contexts/RoombaContext";
import MoveForwardControl from "./MoveForwardControl";

type TestMoveForwardControlProps = {
  spyMoveForward: () => void;
};

const TestMoveForwardControl = ({
  spyMoveForward,
}: TestMoveForwardControlProps) => {
  const state = useContext(RoombaContext);
  state.moveForward = spyMoveForward;

  return <MoveForwardControl />;
};

describe("<MoveForwardControl />", () => {
  const customRender = (children: ReactNode) => {
    return render(
      <RoombaContextProvider size={3}>{children}</RoombaContextProvider>
    );
  };
  it("SHOULD render", () => {
    const { container } = customRender(<MoveForwardControl />);

    expect(container).toMatchSnapshot();
  });

  it("SHOULD call moveForward WHEN button is clicked", () => {
    const spy = jest.fn();
    customRender(<TestMoveForwardControl spyMoveForward={spy} />);

    const element = screen.getByTestId("move-forward-button");

    fireEvent.click(element);
    expect(spy).toHaveBeenCalledTimes(1);

    fireEvent.click(element);
    expect(spy).toHaveBeenCalledTimes(2);
  });
});
