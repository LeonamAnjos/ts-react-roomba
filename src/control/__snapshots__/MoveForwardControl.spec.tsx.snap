// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoveForwardControl /> SHOULD render 1`] = `
<div>
  <button
    data-testid="move-forward-button"
  >
    Move Forward
  </button>
</div>
`;
